src/screens/signUp/index.js

```diff
@@ -1,67 +1,145 @@
 import React, {PureComponent} from 'react';
-import {View, Button} from '../../components';
+import {View, Button, Touchable, Text} from '../../components';
 import {Input} from 'react-native-elements';
 import {signUp} from '../../store/actions';
 import {connect} from 'react-redux';
 import NavigationService from '../../services/NavigationService';
-import {styles} from './styles'
+import {styles} from './styles';
+import * as values from '../../constants/values';
+import * as schema from '../../constants/validationSchema';
+import i18n from '../../localization';
+
 
 class SignUp extends PureComponent {
+
+  static navigationOptions = {
+    headerLeft: (
+      <Touchable
+        onPress={() => NavigationService.pop(1)}
+        style={styles.backButton}>
+        <Text style={styles.backText}>{i18n.t('BACK_BUTTON')}</Text>
+      </Touchable>
+    ),
+  };
+
   state = {
     username: '',
     email: '',
     password: '',
+    isValid: {usernameError: true, emailError: true, passwordError: true},
   };
-  
+
+  validation = (username, email, password) => {
+    let usernameError = schema.schemaUsername.isValidSync({
```

Comment:
it's so bad, you can write only one schema and pass object here, and set abortEarly property as false